test/php/gutenberg-coding-standards/Gutenberg/Sniffs/Commenting/SinceTagSniff.php

```diff
@@ -0,0 +1,614 @@
+		$matches = array();
+		if ( 1 === preg_match( '/^MU \((?<version>.+)\)/', $version, $matches ) ) {
+			$version = $matches['version'];
+		}
+
+		return version_compare( $version, '0.0.1', '>=' );
+	}
+
+
+	/**
+	 * Returns violation codes for a specific token type.
+	 *
+	 * @param string $token_type The type of token (e.g., Function, Property) to retrieve violation codes for.
+	 * @return array An array containing violation codes for missing since tag, missing version value, and invalid version value.
+	 */
+	protected static function get_violation_codes( $token_type ) {
+		return array(
+			'missing_since_tag'     => 'Missing' . $token_type . 'SinceTag',
+			'missing_version_value' => 'Missing' . $token_type . 'VersionValue',
+			'invalid_version_value' => 'Invalid' . $token_type . 'VersionValue',
+		);
+	}
+
+	/**
+	 * Checks if the provided visibility level is below the set minimum visibility level.
+	 *
+	 * @param string $visibility The visibility level to check.
+	 * @return bool Returns true if the provided visibility level is below the minimum visibility level, false otherwise.
+	 */
+	protected function check_below_minimum_visibility( $visibility ) {
+		if ( 'public' === $this->minimumVisibility && in_array( $visibility, array( 'protected', 'private' ), true ) ) {
+			return true;
+		}
+
+		if ( 'protected' === $this->minimumVisibility && 'private' === $visibility ) {
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Finds the docblock associated with a hook, starting from a specified position in the token stack.
+	 * Since a line containing a hook can include any type of tokens, this method backtracks through the tokens
+	 * to locate the first token on the current line. This token is then used as the starting point for searching the docblock.
+	 *
+	 * @param File $phpcs_file    The file being scanned.
+	 * @param int  $stack_pointer The position to start looking for the docblock.
+	 * @return array|false An associative array containing the start and end tokens of the docblock, or false if not found.
+	 */
+	protected static function find_hook_docblock( File $phpcs_file, $stack_pointer ) {
+		$tokens       = $phpcs_file->getTokens();
+		$current_line = $tokens[ $stack_pointer ]['line'];
+
+		for ( $i = $stack_pointer; $i >= 0; $i-- ) {
+			if ( $tokens[ $i ]['line'] < $current_line ) {
+				// The previous token is on the previous line, so the current token is the first on the line.
+				return static::find_docblock( $phpcs_file, $i + 1 );
+			}
+		}
+
+		return static::find_docblock( $phpcs_file, 0 );
+	}
+
+	/**
+	 * Determines if a T_STRING token represents a function call.
+	 *
+	 * @param File $phpcs_file    The file being scanned.
+	 * @param int  $stack_pointer The position to start looking for the docblock.    *
+	 * @return bool True if the token represents a function call, false otherwise.
+	 */
+	protected static function is_function_call( File $phpcs_file, $stack_pointer ) {
```

Comment:
I wish there was a PHPCS or PHPCSUtils method that we could use to check if the token is a function call or not, but apparently, there isn't. At least I was not able to find one when I looked for it the first time I reviewed this PR.

Reply:
This could be done, but let me note that the implementation was largely copied from `PHP_CodeSniffer\Standards\PEAR\Sniffs\Commenting\FunctionCommentSniff::process()`, so it is not new code.